### PR TITLE
TestSentEnvelope destructive test to status-scale

### PR DESCRIPTION
**Pull request by adriacidre**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -43,3 +43,7 @@
 [[constraint]]
   branch = "master"
   name = "github.com/olekukonko/tablewriter"
+
+[[constraint]]
+  name = "github.com/ethereum/go-ethereum"
+  version = "v1.8.5"
```
Comment (dshulyak) [inline Gopkg.toml]: why 1.8.5?

File: tests/peers_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+func TestSentEnvelope(t *testing.T) {
+	// Setup cluster
+	c := ClusterFromConfig()
+	require.NoError(t, c.Create(context.TODO(), cluster.ScaleOpts{Boot: 1, Relay: 2, Deploy: true}))
+	defer c.Clean(context.TODO()) // handle interrupt signal
+
+	var client *rpc.Client
+	// Get the rpc client
+	Eventually(t, func() error {
+		client = c.GetRelay(0).Admin().Client()
```
Comment (dshulyak) [inline tests/peers_test.go]: 1. you shouldnt need `eventually` for that. client is set only when health check for a node as performed
2. please add a separate method for Client. like c.GetRelay(0).Whsper or Client. ideally to wrap it with useful helper, like  Whisper.DebugPost, Whisper.Post

File: tests/peers_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	var hash common.Hash
+	msg := whisperMsg(t, client)
+
+	// With default network capabilities we should be able to succesfully send
+	// messages
+	assert.NoError(t, client.Call(&hash, "debug_postSync", msg))
+
+	// Switching connected peers network connection
+	require.NoError(t, c.GetRelay(0).EnableConditions(context.TODO(), network.Options{
```
Comment (dshulyak) [inline tests/peers_test.go]: did you repeat those lines intentionally or by mistake? i dont understand why you need to block local address

File: tests/peers_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	require.NoError(t, c.GetRelay(1).EnableConditions(context.TODO(), network.Options{
+		PacketLoss: 100,
+		TargetAddr: strings.Split(peers[0].Network.RemoteAddress, ":")[0],
+	}))
+
+	// As our current node doesn't have any connected peer, no message should be sent
+	assert.Error(t, client.Call(&hash, "debug_postSync", msg))
+}
+
+func whisperMsg(t *testing.T, client *rpc.Client) whisperv6.NewMessage {
```
Comment (dshulyak) [inline tests/peers_test.go]: i suggested above that it makes sense to make a wrapper similar to Admin. this wrapper will have 2 methods for now

```
NewSymKey(context.Context) (string, err)
PostDebug(context.Context, whisperv6.Message) (hash, error)
```

File: tests/peers_test.go (hunk 3 above)
Comment (adriacidre) [reply tests/peers_test.go]: hey @dshulyak! they're for both (0 & 1) relay peers. What's your suggestion?

File: cluster/admin.go [hunk 5]
```diff
@@ -18,3 +18,7 @@ func (a Admin) Self(ctx context.Context) (rst *p2p.NodeInfo, err error) {
 func (a Admin) Peers(ctx context.Context) (rst []*p2p.PeerInfo, err error) {
 	return rst, a.client.CallContext(ctx, &rst, "admin_peers")
 }
+
+func (a Admin) Client() *rpc.Client {
```
Comment (dshulyak) [inline cluster/admin.go]: remove please

**Comment by adriacidre** [comment]
ok, all fixed now :)

File: whisper/whisper.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+type Client struct {
+	rpcClient *rpc.Client
+}
+
+func New(c *rpc.Client) *Client {
+	return &Client{rpcClient: c}
+}
+
+func (c *Client) Post(hash *common.Hash, msg whisperv6.NewMessage) error {
```
Comment (dshulyak) [inline whisper/whisper.go]: it doesnt make sense to use pointers in this client interface

File: whisper/whisper.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	return c.rpcClient.Call(hash, "shhext_postSync", msg)
```
Comment (dshulyak) [inline whisper/whisper.go]: sshext_postSync?

File: tests/peers_test.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+func TestSentEnvelope(t *testing.T) {
+	// Setup cluster
+	c := ClusterFromConfig()
+	require.NoError(t, c.Create(context.TODO(), cluster.ScaleOpts{Boot: 1, Relay: 2, Deploy: true}))
+	defer c.Clean(context.TODO()) // handle interrupt signal
+
+	// Get the rpc client
+	client := c.GetRelay(0).Whisper()
+	assert.NotNil(t, client)
```
Comment (dshulyak) [inline tests/peers_test.go]: assert allows test to continue execution

File: tests/peers_test.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+			return err
+		}
+		if len(peers) != 1 {
+			return fmt.Errorf("peers %+v expected to be %d", peersToIPs(peers), 2)
+		}
+		return nil
+	}, 30*time.Second, 1*time.Second)
+
+	var symID string
+	assert.NoError(t, client.NewSymKey(&symID))
```
Comment (dshulyak) [inline tests/peers_test.go]: same

**Comment by adriacidre** [comment]
@dshulyak I think everything is ready. Should I also remove `TestSentEnvelope` from status-go?

**Pull request merged**